### PR TITLE
Update to Jupyter Packaging 0.10

### PR DESCRIPTION
**Pull request by jtpio**
Similar to the update in the TypeScript cookiecutter: https://github.com/jupyterlab/extension-cookiecutter-ts/pull/143/files

We might also want to update `widgetnbextension`:

 https://github.com/jupyter-widgets/ipywidgets/blob/e0d41f6f02324596a282bc9e4650fd7ba63c0004/widgetsnbextension/pyproject.toml#L4

And: https://github.com/jupyter-widgets/ipywidgets/blob/master/widgetsnbextension/setup.py

Originally brought in https://github.com/jupyter-widgets/ipywidgets/pull/3185#pullrequestreview-650731089

### CODE REVIEW

**Comment by jtpio** [comment]
Need to check whether the test failure is related to https://github.com/jupyterlab/extension-cookiecutter-ts/issues/146

**Comment by martinRenou** [comment]
We'll need to do that on all other widgets

**Comment by jtpio** [comment]
Updating to `0.10` which added the `skip-if-exists` option back.

**Review by bsyouness — APPROVED** [review]
Looks good to me! Are we waiting for someone else to review this before merging?

**Comment by jasongrout** [comment]
Closing and reopening again to trigger the tests

File: widgetsnbextension/setup.py
Comment (jasongrout) [inline widgetsnbextension/setup.py:33]: @jtpio - I've been wondering about how jupyter_packaging handles data files that don't exist until after the build process. Part of the reason for the complicated build process before was that the data files could not be calculated until after the js was built. How are the data files retrieved with `get_data_files` before they are built now?

**Pull request merged**
